src/App.tsx

```diff
@@ -1,26 +1,168 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { wait } from './utils/fetchClient';
+import { UserWarning } from './UserWarning';
+import { Footer } from './components/Footer';
+import { creatTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { getFilterTodos } from './utils/getFilterTodos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Default);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
+
+  const focusInput = useRef<HTMLInputElement>(null);
+
+  const isErrorAddTodo = errorMessage === Errors.Add;
+
+  const handleClearError = () => setErrorMessage(Errors.Default);
+
+  const handleError = (error: Errors) => {
+    setErrorMessage(error);
+
+    wait(3000).then(() => handleClearError());
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleError(Errors.Load);
+      });
+  }, []);
+
+  useEffect(() => {
```

Comment:
why does it depend on todos.length?

Reply:
Because when you update the todo, you don't need to focus on the input.